### PR TITLE
added Nokia SR OS device driver (MD-CLI mode)

### PR DESCRIPTION
**Pull request by hellt**
Hi @ogenstad 
this PR adds basic support for Nokia SR OS driver based on MD-CLI.

### CODE REVIEW

File: pkg/netrasp/sros.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"strings"
+)
+
+// SROS is the Netrasp driver for Nokia SR OS (MD-CLI).
+type sros struct {
+	Connection connection
+}
+
+// Close connection to device.
+func (i sros) Close(ctx context.Context) error {
```
Comment (ogenstad) [inline pkg/netrasp/sros.go:16]: Perhaps change from func (i sros) to func (s sros)?

File: pkg/netrasp/sros.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		return "", fmt.Errorf("unable to enter exclusive edit mode: %w", err)
+	}
+	for _, command := range commands {
+		result, err := i.Run(ctx, command)
+		if err != nil {
+			return output, fmt.Errorf("unable to run command '%s': %w", command, err)
+		}
+		output += result
+	}
+	_, err = i.Run(ctx, "commit")
```
Comment (ogenstad) [inline pkg/netrasp/sros.go:36]: I seem to remember from Alcatel Timos that there was some "root" command, would anything like that be needed to return to the correct context or anything like that?

File: pkg/netrasp/sros.go (hunk 1 above)
Comment (hellt) [reply pkg/netrasp/sros.go]: done in 133b5d7

File: pkg/netrasp/sros.go (hunk 2 above)
Comment (hellt) [reply pkg/netrasp/sros.go]: thanks for reminding me that I technically need to exit from configuration mode to return to the running datastore
this added in 4ee60cb

**Comment by ogenstad** [comment]
Thanks!

**Pull request merged**